consensus/enclave/impl/src/lib.rs

```diff
@@ -581,26 +570,50 @@ impl ConsensusEnclave for SgxConsensusEnclave {
     }
 }
 
-/// Creates a single output belonging to the fee recipient account.
+/// Creates a single output belonging to a specific recipient account.
+/// The output is created using a predictable private key that is derived from the input parameters.
 ///
 /// # Arguments:
-/// * `tx_private_key` - Transaction key used to output the aggregate fee.
-/// * `total_fee` - The sum of all fees in the block.
-fn mint_aggregate_fee(
-    fee_recipient: &PublicAddress,
-    tx_private_key: &RistrettoPrivate,
-    total_fee: u64,
+/// * `recipient` - The recipient of the output.
+/// * `domain_tag` - Domain separator for hashing the input parameters.
+/// * `parent_block` - The parent block.
+/// * `transactions` - The transactions that are included in the current block.
+/// * `amount` - Output amount.
+fn mint_output<T: Digestible>(
+    recipient: &PublicAddress,
+    domain_tag: &'static [u8],
+    parent_block: &Block,
+    transactions: &[T],
```

Comment:
The idea here is that for fee outputs, this will be the list of transactions (`Tx` objects) that are part of the block.
For minted outputs (for the new token types) this will be the list of minting transactions, which will likely only ever be one per block per token type.

I think this is fine since the only purpose of this is to ensure the generated private key is unique depending on the inputs, and the MintTx will likely include some nonce to prevent replay attacks (submitting the same minting tx multiple times) so the MintTx will be guaranteed to be unique.